### PR TITLE
CASEFLOW-4241 Implement backend logic to handle VSO convert-to-virtual form submission for Hearing (before hearing is scheduled)

### PR DESCRIPTION
**Pull request by j-n-t-l**
Resolves #[CASEFLOW-4241](https://vajira.max.gov/browse/CASEFLOW-4241)
Description
As a Developer, I need for both the fields in the hearings_email_recipients_table to be updated and the convert-to-virtual backend logic to be implemented so that the convert-to-virtual form has the necessary Veteran/POA data available to convert a hearing request to virtual.

Acceptance Criteria

1. hearing_email_recipients_table contains the following columns:
- [ ] id
- [ ] created_at
- [ ] email_address
- [ ] timezone
- [ ] type
- [ ] updated_at
- [ ] appeal_id
- [ ] appeal_type
2. The hearings_email_recipients_table is updated with two new rows when a VSO virtual opt-in is processed
-One row is added/updated to store the following Veteran information 
- [ ] email_address: Veteran/Appellant email (entered on convert-to-virtual form)
- [ ] timezone: Veteran/Appellant timezone (entered on convert-to-virtual form)
- [ ] type:  AppellantHearingEmailRecipient
- [ ] appeal_id
- [ ] appeal_type
-One row is added/updated to store the following POA information
- [ ] email_address: POA/Representative email
- [ ] timezone: POA/Representative timezone (entered on convert-to-virtual form)
- [ ] type: RepresentativeHearingEmailRecipient
- [ ] appeal_id
- [ ] appeal_type

### CODE REVIEW

File: app/models/tasks/change_hearing_request_type_task.rb [hunk 1]
```diff
@@ -61,7 +61,10 @@ def update_from_params(params, user)
 
     if payload_values&.[](:changed_hearing_request_type).present?
       update_appeal_and_self(payload_values, params)
-
+      # FIX ME make sure task is actually assigned to the user as well
+      if user.can?("VSO")
```
Comment (ksross-va) [inline app/models/tasks/change_hearing_request_type_task.rb:65]: I think we should change this to check for the existence of email_recipients in the params rather than checking for a VSO user. Currently they correspond, but that could change in the future and what we really care about is updating email_recipients if we have them.

File: app/models/tasks/change_hearing_request_type_task.rb [hunk 2]
```diff
@@ -102,6 +105,46 @@ def update_appeal_and_self(payload_values, params)
...
+        appeal: appeal
+      )
+    else
+      rep_recipient.update!(
+        timezone: payload_values[:email_recipients][:appellant_tz]
+      )
+    end
+  end
+
+  def update_hearing_email_recipients(payload_values, params)
```
Comment (ksross-va) [inline app/models/tasks/change_hearing_request_type_task.rb:141]: We should consider continuing the existing naming convention of `create_or_update_` rather than just `update_`

**Review by ksross-va — CHANGES_REQUESTED** [review]
Remove git-fresh and local/vacols/local

File: app/models/tasks/change_hearing_request_type_task.rb (hunk 1 above)
Comment (j-n-t-l) [reply app/models/tasks/change_hearing_request_type_task.rb]: fixed

File: app/models/tasks/change_hearing_request_type_task.rb (hunk 2 above)
Comment (j-n-t-l) [reply app/models/tasks/change_hearing_request_type_task.rb]: fixed

File: app/models/hearings/hearing_email_recipient.rb [hunk 3]
```diff
@@ -9,11 +9,12 @@ def self.email_error_message
     appellant: "appellant",
     veteran: "veteran",
     representative: "representative",
-    judge: "judge"
+    judge: "judge",
+    vso: "VSO"
```
Comment (ksross-va) [inline app/models/hearings/hearing_email_recipient.rb:13]: VSO is not a recipient role, they fall under "representative." Does this change do anything functionally that I'm missing?

File: app/models/hearings/hearing_email_recipient.rb [hunk 4]
```diff
@@ -9,11 +9,12 @@ def self.email_error_message
...
   }.freeze
 
   ATTRIBUTES = [
-    :id, :timezone, :email_address, :type
+    :id, :timezone, :email_address, :type, :appeal_id, :appeal_type
```
Comment (ksross-va) [inline app/models/hearings/hearing_email_recipient.rb:17]: What is the purpose of this change?

File: app/models/tasks/change_hearing_request_type_task.rb [hunk 5]
```diff
@@ -102,6 +104,58 @@ def update_appeal_and_self(payload_values, params)
...
+      )
+    end
+  end
+
+  def create_or_update_representative_email_recipients(payload_values)
+    rep_recipient = appeal.email_recipients.find_by(type: "RepresentativeHearingEmailRecipient")
+    if rep_recipient.blank?
+      # create HER object for representative
+      RepresentativeHearingEmailRecipient.create!(
+        email_address: "caseflow-representative@test.com",
```
Comment (ksross-va) [inline app/models/tasks/change_hearing_request_type_task.rb:139]: This should be the payload value, not hardcoded

File: app/models/tasks/change_hearing_request_type_task.rb [hunk 6]
```diff
@@ -102,6 +104,58 @@ def update_appeal_and_self(payload_values, params)
...
+        timezone: payload_values[:email_recipients][:representative_tz],
+        appeal: appeal,
+        hearing_type: find_hearing_type
+      )
+    else
+      # update recipient if it already exists in the appeal
+      rep_recipient.update!(
+        timezone: payload_values[:email_recipients][:representative_tz]
```
Comment (ksross-va) [inline app/models/tasks/change_hearing_request_type_task.rb:147]: This should also update the email_address right?

File: app/models/tasks/change_hearing_request_type_task.rb [hunk 7]
```diff
@@ -102,6 +104,58 @@ def update_appeal_and_self(payload_values, params)
...
+      )
+    end
+  end
+
+  def create_or_update_hearing_email_recipients(payload_values, params)
+    create_or_update_appellant_email_recipients(payload_values)
+    create_or_update_representative_email_recipients(payload_values)
+    # idk if this is needed
+    update!(params)
```
Comment (ksross-va) [inline app/models/tasks/change_hearing_request_type_task.rb:156]: This line is unnecessary and can be removed.

**Review by ksross-va — CHANGES_REQUESTED** [review]

**Review by ksross-va — APPROVED** [review]

**Review by ksross-va — APPROVED** [review]
Looks good!
